package.json

```
@@ -1,22 +1,50 @@
 {
-  "name": "post-message-stream",
+  "name": "@metamask/post-message-stream",
   "version": "4.0.0",
   "description": "Sets up a duplex object stream over window.postMessage",
-  "main": "index.js",
+  "publishConfig": {
+    "registry": "https://registry.npmjs.org/",
+    "access": "public"
+  },
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "types": "dist/index.d.ts",
+  "files": [
+    "dist/"
+  ],
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "yarn build && yarn testOnly",
+    "testOnly": "jest --runInBand",
```

Do we need the `--runInBand` flag here?

```suggestion
    "testOnly": "jest",
```

The tests seem to pass without it, and the docs suggest it is a niche use case:

```
  --runInBand, -i               Run all tests serially in the current process
                                (rather than creating a worker pool of child
                                processes that run tests). This is sometimes
                                useful for debugging, but such use cases are
                                pretty rare.                           [boolean]
```

---

this reduces threads to 1 and is recommended for CI (also allows debugging with ndb)